jar/jar.go

```diff
@@ -385,7 +392,9 @@ func (c *checker) checkFile(zf *zip.File, depth int, size int64, jar string) err
 	if err != nil {
 		return fmt.Errorf("open file %s: %v", p, err)
 	}
-	buf, err := readFull(f, fi)
+	buf := dynBufPool.Get().([]byte)
+	buf, err = readFull(f, fi, buf)
+	defer dynBufPool.Put(buf)
```

Comment:
Rather than defer this, I think you should recycle it immediately.

Also, could this just be an implementation detail of readFull? it's not used elsewhere.

Reply:
Oops, my bad. It is clearly used below.